Source/WebCore/platform/network/BlobRegistryImpl.cpp

```diff
@@ -392,25 +418,43 @@ Vector<RefPtr<BlobDataFileReference>> BlobRegistryImpl::filesInBlob(const URL& u
     return result;
 }
 
-void BlobRegistryImpl::addBlobData(const String& url, RefPtr<BlobData>&& blobData, const std::optional<WebCore::SecurityOriginData>&)
+void BlobRegistryImpl::addBlobData(const String& url, RefPtr<BlobData>&& blobData, const std::optional<WebCore::SecurityOriginData>& topOrigin)
 {
+    ASSERT(BlobURL::isInternalURL(URL { { }, url }) || topOrigin);
     auto addResult = m_blobs.set(url, WTFMove(blobData));
-    if (addResult.isNewEntry)
+    if (addResult.isNewEntry) {
```

Comment:
Nit: I think we prefer the following pattern to remove nested conditions for better readability:
```
if (!condition) 
    return;
// continue to do work
```
Same for the other functions (unless work is one-line).